sample/trace_key.go

```diff
@@ -40,75 +43,194 @@ func newTraceKey(fields []string, useTraceLength bool) *traceKey {
 		fields:         nonRootFields,
 		rootOnlyFields: rootOnlyFields,
 		useTraceLength: useTraceLength,
+		distinctValue:  &distinctValue{buf: make([]byte, 0, 1024)},
+		keyBuilder:     &bytes.Buffer{},
 	}
 }
 
 // build, builds the trace key based on the configuration of the traceKeyGenerator
 // returns the number of values used to build the key
 func (d *traceKey) build(trace *types.Trace) (string, int) {
 	fieldCount := 0
-	// fieldCollector gets all values from the fields listed in the config, even
-	// if they happen multiple times.
-	fieldCollector := make(map[string][]string)
 
 	// for each field, for each span, get the value of that field
 	spans := trace.GetSpans()
-	uniques := generics.NewSetWithCapacity[string](maxKeyLength)
+	uniques := d.distinctValue
+	uniques.init(d.fields, maxKeyLength)
+	defer func() {
+		// reset the distinctValue for reuse
+		uniques.Reset()
```

Comment:
Seems like this should be implicit with init()

A convention I see a lot is to have Reset() take arguments to set up for the new task in one combined operation.